### PR TITLE
Adds spell power upgrades for fireball, shoe snatching charm and summon robes. Spell channeling for wizard spells

### LINKED ISSUE #11096: Clown Curse lacks target selection
When you have two people next to you, the game will randomly select someone for your clown fun.
Unfortunately this also means that those already clowned can be hit again, completely wasting the spell if you're in a tight situation.

I beg that it either get:
Target selection
or will not target people who have already been clowned

### PR DESCRIPTION
**Pull request by unid15**
- Upgrading fireball's power turns it into Controlled Fireball. Controlled Fireball flies at whatever you click, instead of whatever's in front of you.

Also fixed a horrible fireball runtime

faerbol
- Upgrading shoe snatching's charm power turns it into Shoe Snatching Scourge. When cast on somebody who isn't wearing shoes, it will summon 4 glass shards around them.

shoes (gif is outdated, shards only summon if victim isn't wearing shoes)
- Upgrading summon robes' power turns it into Summon Hardsuit. It summons a gem-encrusted hardsuit, breath mask and an oxygen tank.

(to prevent exploits, summon robes is no longer castable in the wizard's den)

robes

Many spells now use channeling (you have to click on the spell and then on your target) - Fixes #11096

### CODE REVIEW

**Comment by clusterfack** [comment]
Channeling is indeed the best

**Comment by ScrambledSteak** [comment]
Right after I make the spell tier list, you go and change shit to mess with it
thanks

**Comment by AutismSpider** [comment]
butt bot spell going use channeling i hope.

**Comment by Intigracy** [comment]
> (to prevent exploits, summon robes is no longer castable in the wizard's den)

I was about to ask about this, then read the comment.

It's to prevent you from summoning a hardsuit and then refunding the spell.

File: code/modules/spells/targeted/projectile/fireball.dm [hunk 1]
```diff
@@ -25,12 +25,50 @@
 	var/ex_light = 2
 	var/ex_flash = 5
 
+	level_max = list(Sp_TOTAL = 7, Sp_SPEED = 4, Sp_POWER = 1)
+
+	var/list/explosion_by_power = list( //Explosion stats corresponding to the spell's power level. Level 1 explosion is the first list (-1,1,2), level 2 is the second, ...
+		list(-1, 1, 2, 5)
+	)
```
Comment (Intigracy) [inline code/modules/spells/targeted/projectile/fireball.dm]: did you mean to not include more power levels?

**Comment by Kurfursten** [comment]
I'm not sure about Shoe Snatching Scourge. With no cast delay (like Goliath's tentacles), and a 100% chance that the direction you were moving in will have a shard in it, this seems like it will work functionally as an unlimited use egun: click the target and they fall over, taking damage. The only case this wouldn't happen is if the target is already standing perfectly still so you can fireball them. Now I'm not totally sure because it has been a while since I've looked at spellcode, but I think Shoe Snatch can be cast through walls, too, if you have the Orb, which means murderboning is even easier.

Shoe snatch is already a pretty strong spell to a strategic wizard that leaves a lot of shards around.

**Comment by 9600bauds** [comment]
Sounds hilarious
Let's leave it in and see if anyone abuses it

File: code/modules/spells/targeted/projectile/fireball.dm (hunk 1 above)
Comment (unid15) [inline code/modules/spells/targeted/projectile/fireball.dm]: Yep, originally I made the explosion bigger with every power upgrade but then I decided to do something else

**Comment by unid15** [comment]
In theory iit'sbalanced by the fact that it costs 2 spell points, and it does nothing against immobile targets and dionae

Let's see how strong it is in practice. If it's too powerful, it can always be nerfed

**Comment by AutismSpider** [comment]
Don't need the orb for Shoe Snatch old one any way it works past walls.

This new one will need the orb to shoe snatch got click on them and shit.

**Comment by mph55** [comment]
Couldn't you keep the old fireball functionality too? The two seconds fireball spam was a great gimmick.

**Comment by Shadowmech88** [comment]
Shoe snatch only works if the person is actually wearing shoes, right? Couldn't the scourge spell be countered by not wearing shoes in the first place? Would the shards still get spawned?

**Comment by Kurfursten** [comment]
No, it's an if branch without an exit clause. Shards will still spawn.

File: code/modules/spells/targeted/projectile/fireball.dm (hunk 1 above)
Comment (Intigracy) [inline code/modules/spells/targeted/projectile/fireball.dm]: Then why not remove this if it's unused now?

**Comment by Shadowmech88** [comment]
Then maybe the solution is to make shards not spawn if the victim isn't wearing shoes to steal.

That way it'll still mess with a majority of people, but it can still be countered by people who think to remove their own shoes.

**Comment by unid15** [comment]
Nerfed it, now it only spawns glass if the victim is wearing shoes

**Comment by Shadowmech88** [comment]
As an alternative solution that I just thought of, you could also make it so that the glass shards are only spawned when the victim is _not_ wearing shoes.
This would mean that it'd take two casts to be a threat to the average crewmember, but the crew would also have to constantly be replacing their shoes since just giving up and walking around barefoot makes them vulnerable to the scourge effect.

Whichever you think would be best.

Personally I'm leaning toward this idea since with the previous solution, someone could just call out the fact that the wizard was using scourge and everyone would remove their shoes, which would prevent the wizard both from stealing their shoes and spawning shards, making the spell effectively useless.

File: code/modules/spells/targeted/equip/robesummon.dm [hunk 2]
```diff
@@ -36,7 +40,24 @@
...
+
+	usr.visible_message("<span class='danger'>[user] puts on \his robe and wizard hat!</span>", \
 						"<span class='danger'>You put on your robe and wizard hat!</span>")
 
-	..()
\ No newline at end of file
+	..()
+
+/spell/targeted/equip_item/robesummon/empower_spell()
+	if(!valid_outfits.Find("suit"))
```
Comment (ComicIronic) [inline code/modules/spells/targeted/equip/robesummon.dm]: Use defines for these strings.

File: code/modules/spells/targeted/projectile/fireball.dm [hunk 3]
```diff
@@ -25,12 +25,50 @@
...
 
 /spell/targeted/projectile/dumbfire/fireball/prox_cast(var/list/targets, spell_holder)
 	for(var/mob/living/M in targets)
 		apply_spell_damage(M)
 	explosion(get_turf(spell_holder), ex_severe, ex_heavy, ex_light, ex_flash)
+	return targets
+
+/spell/targeted/projectile/dumbfire/fireball/empower_spell()
+	spell_levels[Sp_POWER]++
+	var/power_level = min(spell_levels[Sp_POWER], 3)
```
Comment (ComicIronic) [inline code/modules/spells/targeted/projectile/fireball.dm]: Why 3?

**Comment by unid15** [comment]
Fixed the issues, changed Shoe Snatching Scourge again (implementing @Shadowmech88's suggestion) - it will now only summon glass shards if the victim isn't wearing shoes.

**Comment by ghost** [comment]
No shards. It's a guaranteed kill while you switch between sliding them into an adj shard and waiting to bump into the opposing if they don't go for that pickup, stabbing them while the shard stun keeps them helpless, and just murdering them with projectiles while they're helpless to juke or flashbang. x-ray and jaunt/false walls is gonna let you get a lot of ggnore kills and even then a 15 second sax isn't a high bar to end someone even without dropping cleaner grenades or snatching from inside locker/disposals.

**Comment by Shadowmech88** [comment]
If the people don't want to get sharded, they'd better make sure to carry a lot of backup shoes.

**Comment by unid15** [comment]
It only summons glass if you weren't wearing shoes, so if you get sharded it's your fault - you either didn't bother to retreat and replace your shoes, or you're a tajaran

**Pull request merged**